alica_engine/include/engine/Output.h

```diff
@@ -4,19 +4,25 @@
 
 #include <iterator>
 #include <ostream>
+#include <type_traits>
 
 namespace alica
 {
 
-inline std::ostream& operator<<(std::ostream& out, const AgentGrp& ag)
+template <typename T, std::enable_if_t<std::is_pointer<T>::value, bool> = true>
+std::ostream& operator<<(std::ostream& out, const std::vector<T>& gr)
 {
-    std::copy(ag.begin(), ag.end(), std::ostream_iterator<AgentId>(out, " "));
+    for (const auto* const v : gr) {
+        out << *v << " ";
```

Comment:
Risky since can be null?

Reply:
potentially it could be, I guess. Will change